<h2>ACTIVITY 2 - UK ROAD ACCIDENT</h2>
<hr>

<h4>Analyst : Mark Jayson D. Ramilo</h4>
<hr>

<h2>1.) IMPORT NECESSARY LIBRARIES</h2>
<hr>

In [1]:
#import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h2>2.) CONVERT DATASET TO DATAFRAME</h2>
<hr>

In [2]:
accident = pd.read_csv("C:\\Users\\DESIREE\\Downloads\\uk_road_accident (1).csv")
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<H3>3.) CHECK DATAFRAME AND INFORMATION</H3>
<hr>

In [3]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<H2>4.) CHECK BASIC DESCRIPTIVE STATISTICS</H2>
<hr>

In [4]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

<h2>5.) CALL EACH COLUMN INDIVIDUALLY</h2>
<hr>

In [5]:
accident['Index']

0         200701BS64157
1         200701BS65737
2         200701BS66127
3         200701BS66128
4         200701BS66837
              ...      
660674    201091NM01760
660675    201091NM01881
660676    201091NM01935
660677    201091NM01964
660678    201091NM02142
Name: Index, Length: 660679, dtype: object

In [6]:
accident['Accident_Severity']

0         Serious
1         Serious
2         Serious
3         Serious
4          Slight
           ...   
660674     Slight
660675     Slight
660676     Slight
660677    Serious
660678    Serious
Name: Accident_Severity, Length: 660679, dtype: object

In [7]:
accident['Accident Date']

0           5/6/2019
1           2/7/2019
2         26-08-2019
3         16-08-2019
4           3/9/2019
             ...    
660674    18-02-2022
660675    21-02-2022
660676    23-02-2022
660677    23-02-2022
660678    28-02-2022
Name: Accident Date, Length: 660679, dtype: object

In [8]:
accident['Latitude']

0         51.506187
1         51.495029
2         51.517715
3         51.495478
4         51.488576
            ...    
660674    57.374005
660675    57.232273
660676    57.585044
660677    57.214898
660678    57.575210
Name: Latitude, Length: 660679, dtype: float64

In [9]:
accident['Light_Conditions']

0               Darkness - lights lit
1                            Daylight
2         Darkness - lighting unknown
3                            Daylight
4               Darkness - lights lit
                     ...             
660674                       Daylight
660675         Darkness - no lighting
660676                       Daylight
660677         Darkness - no lighting
660678                       Daylight
Name: Light_Conditions, Length: 660679, dtype: object

In [10]:
accident['District Area']

0         Kensington and Chelsea
1         Kensington and Chelsea
2         Kensington and Chelsea
3         Kensington and Chelsea
4         Kensington and Chelsea
                   ...          
660674                  Highland
660675                  Highland
660676                  Highland
660677                  Highland
660678                  Highland
Name: District Area, Length: 660679, dtype: object

In [11]:
accident['Longitude']

0        -0.209082
1        -0.173647
2        -0.210215
3        -0.202731
4        -0.192487
            ...   
660674   -3.467828
660675   -3.809281
660676   -3.862727
660677   -3.823997
660678   -3.895673
Name: Longitude, Length: 660679, dtype: float64

In [12]:
accident['Number_of_Casualties']

0         1
1         1
2         1
3         1
4         1
         ..
660674    2
660675    1
660676    1
660677    1
660678    1
Name: Number_of_Casualties, Length: 660679, dtype: int64

In [13]:
accident['Number_of_Vehicles']

0         2
1         2
2         3
3         4
4         2
         ..
660674    1
660675    1
660676    3
660677    2
660678    1
Name: Number_of_Vehicles, Length: 660679, dtype: int64

In [14]:
accident['Road_Surface_Conditions']

0                  Dry
1          Wet or damp
2                  Dry
3                  Dry
4                  Dry
              ...     
660674             Dry
660675    Frost or ice
660676    Frost or ice
660677     Wet or damp
660678     Wet or damp
Name: Road_Surface_Conditions, Length: 660679, dtype: object

In [15]:
accident['Road_Type']

0         Single carriageway
1         Single carriageway
2                        NaN
3         Single carriageway
4                        NaN
                 ...        
660674    Single carriageway
660675    Single carriageway
660676    Single carriageway
660677    Single carriageway
660678      Dual carriageway
Name: Road_Type, Length: 660679, dtype: object

In [16]:
accident['Urban_or_Rural_Area']

0         Urban
1         Urban
2         Urban
3         Urban
4         Urban
          ...  
660674    Rural
660675    Rural
660676    Rural
660677    Rural
660678    Rural
Name: Urban_or_Rural_Area, Length: 660679, dtype: object

In [17]:
accident['Weather_Conditions']

0            Fine no high winds
1         Raining no high winds
2                           NaN
3            Fine no high winds
4                           NaN
                  ...          
660674       Fine no high winds
660675       Fine no high winds
660676       Fine no high winds
660677       Fine no high winds
660678    Snowing no high winds
Name: Weather_Conditions, Length: 660679, dtype: object

In [18]:
accident['Vehicle_Type']

0                                          Car
1                                          Car
2                        Taxi/Private hire car
3         Bus or coach (17 or more pass seats)
4                                Other vehicle
                          ...                 
660674                                     Car
660675                                     Car
660676                                     Car
660677                   Motorcycle over 500cc
660678                                     Car
Name: Vehicle_Type, Length: 660679, dtype: object

<h2>6.) CHECK FOR NULL VALUES</h2>

In [19]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h2>7.) FILL NULL VALUES WITH STATISTCAL VALUES</h2>
<hr>

In [20]:
accident['Latitude'] = accident['Latitude'] .fillna(accident['Latitude'].mean())
accident['Longitude'] = accident['Longitude'] .fillna(accident['Longitude'].mean())
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'] .fillna(accident['Road_Surface_Conditions'].mode()[0])
accident['Road_Type'] = accident['Road_Type'] .fillna(accident['Road_Type'].mode()[0])
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'] .fillna(accident['Urban_or_Rural_Area'].mode()[0])
accident['Weather_Conditions'] = accident['Weather_Conditions'] .fillna(accident['Weather_Conditions'].mode()[0])

In [21]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h2>8.) CHECK COLUMN DATA TYPES</h2>
<hr>

In [22]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660679 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660679 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  660679 non-null  object 
 10  Road_Type                660679 non-null  object 
 11  Urban_or_Rural_Area      660679 non-null  object 
 12  Weather_Conditions       660679 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [23]:
accident['Index'] = accident ['Index'].astype('category')

In [24]:
accident['Accident_Severity'] = accident ['Accident_Severity'].astype('category')

In [25]:
accident['Accident Date'] = accident ['Accident Date'].astype('category')

In [26]:
accident['Light_Conditions'] = accident ['Light_Conditions'].astype('category')

In [27]:
accident['District Area'] = accident ['District Area'].astype('category')

In [28]:
accident['Road_Surface_Conditions'] = accident ['Road_Surface_Conditions'].astype('category')

In [29]:
accident['Road_Type'] = accident ['Road_Type'].astype('category')

In [30]:
accident['Urban_or_Rural_Area'] = accident ['Urban_or_Rural_Area'].astype('category')

In [ ]:
accident['Weather_Conditions'] = accident ['Weather_Conditions'].astype('category')

In [ ]:
accident['Vehicle_Type'] = accident ['Vehicle_Type'].astype('category')

In [ ]:
accident.info()

<h2>CLEARING ANY INCONSISTENCIES WITH THE DATA SET</h2>
<hr>

In [ ]:
accident['Accident Date'] = accident['Accident Date'].str.strip() #to remove any extra white space
accident['Accident Date'] = accident['Accident Date'].astype('str') #to convert into a string
accident['Accident Date'] = accident['Accident Date'].str.replace('/', '-')

In [ ]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

<h2>ADJUSTING DATA TYPES</h2>
<HR>

In [ ]:
accident.dtypes

<h2>EXTRACTING DATE INFORMATION USING PANDAS DATE TIME</h2>
<hr>

In [ ]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek #Monday=0, Sunday=6

In [ ]:
accident.isnull().sum()

<h2>20 QUESTIONS WITH CODE, ANSWER, AND INSIGHTS...</h2>
<HR>

<h2>QUESTION 1</h2>
<h3>How many total accidents are in the dataset?</h3>
<hr>

In [38]:
total_accidents = len(accident)
total_accidents

660679

<h2>INSIGHT:</h2>
<H3>Based on the dataset, the total of the accident that occured were 660,679.</H3>
<hr>

<h2>QUESTION 2</h2>
<H3>Were there fatal accidents in rural areas?</H3>
<hr>

In [37]:
fatal_rural = accident[(accident["Accident_Severity"] == "Fatal") & (accident["Urban_or_Rural_Area"] == "Rural")]
fatal_rural.shape[0]

5601

<h2>INSIGHT:</h2>
<H3>There were 5,601 fatal accidents in rural areas, which shows that rural crashes tend to be more severe. This could be due to higher speed limits and the longer response time for emergency services.</H3>
<hr>

<H2>QUESTION 3</H2>
<H3>Do accidents in rural areas usually involve more than one vehicle?</H3>
<hr>

In [36]:
rural_multi = accident[(accident["Urban_or_Rural_Area"] == "Rural") & (accident["Number_of_Vehicles"] > 1)]
rural_multi.shape[0]

163235

<H2>INSIGHT:</H2>
<H3>Yes, there were 163,235 rural accidents in daylight, showing that good visibility does not prevent crashes. Many rural accidents are linked to speed and long travel distances.</H3>
<hr>

<h2>QUESTION 4</h2>
<H3>Is it common for accidents with more than 2 casualties to occur at night?</H3>
<hr>

In [35]:
night_cas2 = accident[(accident["Light_Conditions"] != "Daylight") & (accident["Number_of_Casualties"] > 2)]
night_cas2.shape[0]


15431

<H2>INSIGHT:</H2>
<H3>We can observe 15,431 nighttime accidents with more than 2 casualties, highlighting how darkness raises risks for groups of people. Poor visibility and fatigue may worsen the effects of accidents.</H3>
<hr>

<H2>QUESTION 5</H2>
<H3>Does rain affect the likelihood of accidents with more than 1 vehicle?</H3>
<hr>

In [34]:
accident["Poor_Light"] = accident["Light_Conditions"].str.contains("Dark|Dusk|Dawn")
accident[["Poor_Light","Number_of_Casualties"]].corr()


Poor_Light  Number_of_Casualties
Poor_Light              1.000000              0.033687
Number_of_Casualties    0.033687              1.000000

<H2>INSIGHT:</H2>
<H3>The correlation between poor light and casualties is very low (0.034). This suggests poor lighting has minimal direct impact on accident severity.</H3>
<hr>

<H2>QUESTION 6</H2>
<H3>Are fatal accidents more often reported in rural areas than in urban ones?</H3>
<hr>

In [39]:
fatal_rural = accident[(accident["Accident_Severity"] == "Fatal") & (accident["Urban_or_Rural_Area"] == "Rural")]
fatal_rural.shape[0]


5601

<H2>INSIGHT:</H2>
<H3>The dataset shows also that 5,601 fatal accidents were reported in rural areas, highlighting that severe crashes are more common outside urban centers. This pattern may be linked to faster driving speeds and longer emergency response times in rural environments.</H3>
<hr>

<H2>QUESTION 7</H2>
<H3>Are urban areas more prone to multi-vehicle accidents?</H3>
<hr>

In [40]:
urban_group = accident.groupby("Urban_or_Rural_Area")["Number_of_Vehicles"].mean()
urban_group["Urban"]



np.float64(1.811384516147392)

<H2>INSIGHT:</H2>
<H3>The dataset shows that the average number of vehicles involved in urban accidents is approximately 1.81, indicating that most accidents in cities involve one or two vehicles. This suggests that while multi-vehicle collisions occur, single-vehicle or two-vehicle accidents are still the most common in urban areas.</H3>
<hr>

<H2>QUESTION 8</H2>
<H3>Is it frequent that daylight accidents involve more than 3 casualties?</H3>
<hr>

In [41]:
day_cas3 = accident[(accident["Light_Conditions"] == "Daylight") & (accident["Number_of_Casualties"] > 3)]
day_cas3.shape[0]


11822

<H2>INSIGHT:</H2>
<H3>Yes, there are 11,822 accidents that occurred in daylight with more than three casualties, indicating that even under well-lit conditions, severe accidents are still relatively common. This suggests that factors other than lighting, such as traffic density or driver behavior, likely play a significant role in multi-casualty incident.</H3>
<hr>

<H2>QUESTION 9</H2>
<H3>Do motorways show accidents with more than 4 vehicles involved?</H3>
<hr>

In [43]:
motorway_multi4 = accident[(accident["Road_Type"] == "Motorway") & (accident["Number_of_Vehicles"] > 4)]
motorway_multi4.shape[0]



0

<H2>INSIGHT:</H2>
<H3>No accidents on motorways involved more than four vehicles, indicating that large multi-vehicle collisions are extremely rare on these roads. This could be due to effective traffic management, wider lanes, or stricter regulations that help prevent major pile-ups.</H3>
<hr>

<h2>QUESTION 10</h2>
<H3>Which vehicle type and agriculture vehicle is the most accident in fatal accident?</H3>
<hr>

In [42]:
fatal_accidents = accident[accident["Accident_Severity"] == "Fatal"]
fatal_vehicle_count = fatal_accidents.groupby("Vehicle_Type")["Index"].count()
fatal_vehicle_count

Vehicle_Type
Agricultural vehicle                       21
Bus or coach (17 or more pass seats)      325
Car                                      6577
Goods 7.5 tonnes mgw and over             216
Goods over 3.5t. and under 7.5t            67
Minibus (8 - 16 passenger seats)           29
Motorcycle 125cc and under                189
Motorcycle 50cc and under                  95
Motorcycle over 125cc and up to 500cc     105
Motorcycle over 500cc                     339
Other vehicle                              70
Pedal cycle                                 6
Taxi/Private hire car                     155
Van / Goods 3.5 tonnes mgw or under       467
Name: Index, dtype: int64

<h2>INSIGHT:</h2>
<h3>The dataset shows that cars are the vehicle type most involved in fatal accidents, reflecting their prevalence on the roads. This indicates that higher exposure and frequency of car usage significantly increase the likelihood of fatalities compared to less common vehicles.</h3>
<hr>

<H2>QUESTION 11</H2>
<H3>How many accidents occurred during daylight versus night?</H3>
<hr>

In [44]:
light_counts = accident.groupby("Light_Conditions").size()
light_counts

Light_Conditions
Darkness - lighting unknown      6484
Darkness - lights lit          129335
Darkness - lights unlit          2543
Darkness - no lighting          37437
Daylight                       484880
dtype: int64

<h2>INSIGHT:</h2>
<H3>It shows that the majority of accidents occur during daylight (484,880), reflecting higher traffic volumes during the day. However, a significant number of accidents also happen in darkness, especially when lights are lit (129,335), highlighting that nighttime driving still carries substantial risk and that proper lighting alone does not eliminate accidents.</H3>
<hr>

<H2>QUESTION 12</H2>
<H3>Are minibuses more involved in rural accidents?</H3>
<hr>

In [45]:
minibus_area_mean = accident[accident["Vehicle_Type"]=="Minibus (8 - 16 passenger seats)"].groupby("Urban_or_Rural_Area")["Number_of_Vehicles"].mean()
minibus_area_mean


Urban_or_Rural_Area
Rural          1.818942
Unallocated         NaN
Urban          1.795707
Name: Number_of_Vehicles, dtype: float64

<h2>INSIGHT:</h2>
<H3>Minibuses in rural areas have a slightly higher average number of vehicles involved in accidents (1.82) compared to urban areas (1.80). This suggests that rural roads may contribute to slightly more multi-vehicle incidents for minibuses.</H3>
<hr>

<H2>QUESTION 13</H2>
<H3>Are taxis involved in fewer multi-vehicle accidents than cars?</H3>
<hr>

In [46]:
vehicle_multi_mean = accident.groupby("Vehicle_Type")["Number_of_Vehicles"].mean()
vehicle_multi_mean[["Taxi/Private hire car","Car"]]


Vehicle_Type
Taxi/Private hire car    1.836468
Car                      1.832134
Name: Number_of_Vehicles, dtype: float64

<H2>INSIGHT:</H2>
<H3>Taxis have a slightly higher average number of vehicles involved in accidents (1.84) compared to cars (1.83). This suggests both vehicle types are mostly involved in one- or two-vehicle crashes.</H3>
<hr>

<H2>QUESTION 14</H2>
<H3>Are more accidents reported during daylight?</H3>
<hr>

In [47]:
light_mode = accident["Light_Conditions"].mode()[0]
light_mode




'Daylight'

<h2>INSIGHT:</h2>
<H3>Daylight is the most common light condition for accidents. It means that most accidents occur during the day, likely due to higher traffic volume.</H3>
<HR>

<H2>QUESTION 15</H2>
<H3>Do motorcycles contribute to more single-vehicle crashes than cars?</H3>
<HR>

In [48]:
vehicle_group = accident.groupby("Vehicle_Type")["Number_of_Vehicles"].mean()
vehicle_group[["Motorcycle over 500cc","Car"]]



Vehicle_Type
Motorcycle over 500cc    1.834743
Car                      1.832134
Name: Number_of_Vehicles, dtype: float64

<H2>INSIGHT:</H2>
<H3>Motorcycles over 500cc have a slightly higher average number of vehicles involved in accidents (1.83) compared to cars (1.83). This suggests that both are mostly involved in one- or two-vehicle crashes, with minimal difference between them.</H3>
<HR>

<H2>QUESTION 16</H2>
<H3>Which vehicle type causes the most casualties on average?</H3>
<HR>

In [52]:
vehicle_casualties = accident.groupby("Vehicle_Type")["Number_of_Casualties"].mean()
vehicle_casualties



Vehicle_Type
Agricultural vehicle                     1.342065
Bus or coach (17 or more pass seats)     1.349216
Car                                      1.358841
Data missing or out of range             1.500000
Goods 7.5 tonnes mgw and over            1.351881
Goods over 3.5t. and under 7.5t          1.362861
Minibus (8 - 16 passenger seats)         1.345648
Motorcycle 125cc and under               1.332635
Motorcycle 50cc and under                1.337235
Motorcycle over 125cc and up to 500cc    1.343521
Motorcycle over 500cc                    1.359434
Other vehicle                            1.340075
Pedal cycle                              1.370558
Ridden horse                             1.250000
Taxi/Private hire car                    1.368663
Van / Goods 3.5 tonnes mgw or under      1.354537
Name: Number_of_Casualties, dtype: float64

<H2>INSIGHT:</H2>
<H3>As we can see at the data, cars have the highest average number of casualties per accident (1.36) among all vehicle types. This suggests that accidents involving cars tend to result in slightly more injuries than those involving other vehicles.</H3>
<HR>

<h2>QUESTION 17</h2>
<H3>Are buses involved in accidents with more casualties than cars?</H3>
<HR>

In [51]:
bus_car_casualties = accident.groupby("Vehicle_Type")["Number_of_Casualties"].mean()
bus_car_casualties[["Bus or coach (17 or more pass seats)", "Car"]]



Vehicle_Type
Bus or coach (17 or more pass seats)    1.349216
Car                                     1.358841
Name: Number_of_Casualties, dtype: float64

<H2>INSIGHT:</H2>
<H3>Cars have a slightly higher average number of casualties per accident (1.36) compared to buses (1.35). This suggests that despite buses carrying more passengers, car accidents result in slightly more casualties on average.</H3>
<HR>

<H2>QUESITON 18</H2>
<H3>Do foggy conditions cause fewer multi-vehicle accidents than clear weather?</H2>
<HR>

In [54]:
fog_multi = accident.groupby("Weather_Conditions")["Number_of_Vehicles"].mean()
fog_multi[["Fog or mist","Fine no high winds"]]

Weather_Conditions
Fog or mist           1.807823
Fine no high winds    1.835907
Name: Number_of_Vehicles, dtype: float64

<H2>INSIGHT:</H2>
<H3>It shows that accidents in fine weather without high winds involve more vehicles on average (1.84) compared to fog or mist (1.81). This suggests that clear conditions encourage heavier traffic flow, leading to more multi-vehicle crashes.</H3>
<hr>

<h2>QUESTION 19</h2>
<H3>What is the mean number of vehicles per weather condition?.</H3>
<HR>

In [61]:
accident.groupby("Weather_Conditions")["Number_of_Vehicles"].mean()


Weather_Conditions
Fine + high winds        1.779401
Fine no high winds       1.835907
Fog or mist              1.807823
Other                    1.777668
Raining + high winds     1.777951
Raining no high winds    1.831610
Snowing + high winds     1.737853
Snowing no high winds    1.754889
Name: Number_of_Vehicles, dtype: float64

<H2>INSIGHT:</H2>
<H3>Accidents in normal fine weather (with or without winds) tend to involve more vehicles on average than those in snowy conditions. This suggests that clear weather encourages higher traffic flow, which increases the chance of multi-vehicle collisions.</H3>
<HR>

<h2>QUESTION 20</h2>
<H3>Compare average casualties on dry vs wet road surfaces.</H3>
<HR>

In [62]:
accident.groupby("Road_Surface_Conditions")["Number_of_Casualties"].mean()[["Dry","Wet or damp"]]


Road_Surface_Conditions
Dry            1.334545
Wet or damp    1.411064
Name: Number_of_Casualties, dtype: float64

<H2>INSIGHT:</H2>
<H3>Road conditions play a key role in the severity of crashes, with wet surfaces showing a higher casualty average. Even a slight increase highlights how weather-related factors can make roads more dangerous.</H3>
<HR>

<H2>QUESTION 21</H2>
<H3>Correlate district average latitude with the number of accidents.</H3>
<HR>

In [64]:
accident.groupby("District Area")["Latitude"].mean().corr(
    accident.groupby("District Area")["Index"].count())


np.float64(-0.09064804637605312)

<H2>INSIGHT:</H2>
<H3>It seems weak negative correlation suggests that districts located further north tend to have slightly fewer accidents. This pattern may be influenced by lower population density or less traffic volume in northern areas compared to southern ones.</H3>
<hr>

<h2>QUESTION 22</h2>
<H3>Which district area has the most accidents?</H3>
<HR>

In [71]:
accident["District Area"].mode()[0]



'Birmingham'

<H2>INSIGHT:</H2>
<H3>Birmingham stands out as the area with the highest involvement in wet-road crashes, pointing to its heavy traffic flow and possibly dense urban road network that is more vulnerable when surfaces are slippery.</H3>
<HR>

<h2>QUESTION 23</h2>
<H3>Which road surface condition is most frequent for single-vehicle crashes?</H3>
<Hr>

In [69]:
accident[accident["Number_of_Vehicles"]==1]["Road_Surface_Conditions"].mode()[0]

'Dry'

<H2>INSIGHT:</H2>
<H3>The condition for most frequent for single-vehicle crashers is dry, and even when the roads are dry, drivers still experience the most single-vehicle crashes, suggesting that human error or speeding has a stronger impact than weather conditions.</H3>
<hr>

<h2>QUESTION 24</h2>
<H3>Which road type has the highest average number of casualties?</H3>
<HR>

In [74]:
road_avg = accident.groupby("Road_Type")["Number_of_Casualties"].mean()
road_avg[road_avg == road_avg.max()].index[0]


'Dual carriageway'

<h2>INSIGHT:</h2>
<H3>Dual carriageways show the highest average casualties, which reflects how opposing traffic and limited barriers increase crash severity.</H3>
<hr>

<h2>QUESTION 25</h2>
<H3>Which weather condition is most frequent in rural areas?</H3>
<hr>

In [78]:
accident[accident["Urban_or_Rural_Area"]=="Rural"]["Weather_Conditions"].mode()[0]

'Fine no high winds'

<h2>INSIGHT:</h2>
<H3>Rural crashes most often happen during clear weather or fine with no high winds, showing that environment is not the main risk factor, but rather road layout and driving behavior.</H3>
<hr>